sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -403,9 +403,44 @@ std::unique_ptr<RawResponse> CurlTransport::Send(Request& request, Context const
   auto response = session->ExtractResponse();
   // Move the ownership of the CurlSession (bodyStream) to the response
   response->SetBodyStream(std::move(session));
+  // check the consistency of the keep alive headers between the request and the response
+  ValidateKeepAliveHeaders(request, response);
+
   return response;
 }
 
+void CurlTransport::ValidateKeepAliveHeaders(
+    Request& request,
+    std::unique_ptr<RawResponse>& response)
+{
+  // if the server supports keep alive the headers should be present in the response. If they are
+  // they should be the same as the request headers.
+  if (response->GetHeaders().find("Connection") != response->GetHeaders().end()
+      && request.GetHeader("Connection").HasValue()
+      && response->GetHeaders().find("Keep-Alive") != response->GetHeaders().end()
+      && request.GetHeader("Keep-Alive").HasValue()
+      // just in case the server sends the headers in a different case
+      && Azure::Core::_internal::StringExtensions::ToLower(
+             response->GetHeaders().find("Connection")->second)
+          == Azure::Core::_internal::StringExtensions::ToLower(
```

Comment:
Could you cache the responses to `GetHeader` and `GetHeaders` locally so you don't have to constantly go back to the source? `GetHeaders` is completely cheap (it just returns a reference to the map), but `GetHeader` does involve some work.